utils/docker/dev/Dockerfile

```diff
@@ -0,0 +1,62 @@
+# Copyright IBM Corp. All Rights Reserved.
+# Copyright 2020 Intel Corporation
+#
```

Comment:
I want to make the point that we should remove this `dev` image, and leave the `base-dev` one. This requires minor modifications, because both images can `build` FPC, though only `dev` can run all the FPC tests.

The main difference between `base-dev` and `dev`  is the presence of Fabric's source code and the `SGX_MODE` variable in `dev`.
The former, in particular, does not change, since we check out a predefined version.
So we are leaving to the user to do the job (while building the `dev` image), but we can do this before. Additionally, Fabric's code is necessary only because of some tests under `internal`. Otherwise, even `base-dev` can run the FPC tests.
Anyway, without opening a discussion on the tests, I would leave the source code in the image.

The `SGX_MODE` variable instead is set/hardcoded in the image. However, the `dev` image is meant to be used in SIM/HW mode. So, this should not be hardcoded -- besides, we already have `SIM` as default value.
Additionally, any such variable can be set at docker run-time (with the `-e` option). Hence, there is no need to create an image just for setting it. Actually, it would be even desirable to set it at run-time, equal to the value of the host's variable. 

Concluding, `dev` could be removed, and this would not impact the build process. However, for our tests (including local tests in HW mode) to work, `base-dev` must check out Fabric's code, and `make run-dev` (or equivalent target for running `base-dev`) must set the `SGX_MODE` variable appropriately.

Reply:
I think you made a good point. In general I vote for removing the dev image but I still see a good use for it.

Regarding `SGX_MODE` I agree with you and we should set it at docker run-time.

Regarding Fabric, this can also go away once tests are "refactored".

Let me add another difference between `base-dev` and `dev`. The `dev` image may contain custom applications installed via `apt`. The user may set `DOCKER_DEV_IMAGE_APT_ADD_PKGS` to install whatever they need inside the dev container. For instance, I use this feature to install a few things, like `tmux` and `vim` in my dev container, which I will not miss!! :P

Moving `DOCKER_DEV_IMAGE_APT_ADD_PKGS` frpm `dev` to `base-dev` would require to rebuild it from scratch and not leveraging the new docker pull mode. Works but not nice when you just want to have `tmux` installed.
Installing these packages on every startup also not ideal as the container is not persistent.
Let the user create a "custom" image based on `base-dev` that contains additional tools they want, would lead to the solution we already have with dev.

Reply:
> Regarding Fabric, this can also go away once tests are "refactored".

"Away", when tests are refactored -- I guess you know better -- or "in base-dev" at any time to consolidate the two images.

> The user may set DOCKER_DEV_IMAGE_APT_ADD_PKGS ... 

True, but that's not something that impacts FPC: neither the build process, nor the testing (assuming Fabric's code is present, of course).
For this reason, the question is whether FPC should provide this feature, or leave this to the end users/developers. Expectedly, if they need anything other than an apt package, they _can_ and will build their own images, but FPC will not provide any such feature/enablement in dev.
Anyway, fine to keep this for now to avoid any disruption. I have captured this in #616 .